src/antsibull_docs/process_docs.py

```diff
@@ -63,36 +138,45 @@ def normalize_plugin_info(
 
     errors: list[str] = []
     if plugin_type == "role":
+        role_schema: type[BaseModel] = DOCS_SCHEMAS[
+            plugin_type
+        ]  # type: ignore[attr-defined, assignment]
         try:
-            parsed = DOCS_SCHEMAS[plugin_type].parse_obj(plugin_info)  # type: ignore[attr-defined]
-            return parsed.dict(by_alias=True), errors
-        except v1.ValidationError as e:
-            raise ValueError(str(e))  # pylint:disable=raise-missing-from
+            parsed = role_schema.model_validate(plugin_info)
+            return parsed.model_dump(by_alias=True), errors
+        except p.ValidationError as e:
+            raise ValueError(  # pylint:disable=raise-missing-from
+                _exc_to_string(e, role_schema.__name__)
+            )
```

Comment:
Why doesn't this have a `from`?

Reply:
Because we didn't have a `from` in the existing code as well.

Reply:
I guess we can leave it as is and fix it later, but python does generate a confusing chain when you raise an Exception in an Exception handler without a `from`.

Reply:
True. The exceptions here never surface to the user though, since they are caught and their string representation stored.

(Which also means that adding `from` isn't a problem.)